conan/tools/meson/toolchain.py

```diff
@@ -3,187 +3,169 @@
-        self._shared = self._conanfile.options.get_safe("shared")
-        self._fpic = self._conanfile.options.get_safe("fPIC")
-        self._build_env = VirtualBuildEnv(self._conanfile).vars()
-
-        self.definitions = dict()
-        self.preprocessor_definitions = dict()
-
-        def from_build_env(name):
-            return self._to_meson_value(self._build_env.get(name, None))
+        # Values are kept as Python built-ins so users can modify them more easily, and they are
+        # only converted to Meson file syntax for rendering
+        # priority: first user conf, then recipe, last one is default "ninja"
+        backend_conf = conanfile.conf["tools.meson.mesontoolchain:backend"]
+        self._backend = backend_conf or backend or 'ninja'
 
-        self.c = from_build_env("CC")
-        self.cpp = from_build_env("CXX")
-        self.c_ld = from_build_env("CC_LD") or from_build_env("LD")
-        self.cpp_ld = from_build_env("CXX_LD") or from_build_env("LD")
-        self.ar = from_build_env("AR")
-        self.strip = from_build_env("STRIP")
-        self.as_ = from_build_env("AS")
-        self.windres = from_build_env("WINDRES")
-        self.pkgconfig = from_build_env("PKG_CONFIG")
+        build_type = self._conanfile.settings.get_safe("build_type")
+        self._buildtype = {"Debug": "debug",  # Note, it is not "'debug'"
+                          "Release": "release",
+                          "MinSizeRel": "minsize",
+                          "RelWithDebInfo": "debugoptimized"}.get(build_type, build_type)
+        self._b_ndebug = "true" if self._buildtype != "Debug" else "false"
 
+        # https://mesonbuild.com/Builtin-options.html#base-options
+        fpic = self._conanfile.options.get_safe("fPIC")
+        shared = self._conanfile.options.get_safe("shared")
+        self._b_staticpic = fpic if (shared is False and fpic is not None) else None
         # https://mesonbuild.com/Builtin-options.html#core-options
         # Do not adjust "debug" if already adjusted "buildtype"
-        self.buildtype = self._to_meson_build_type(self._build_type) if self._build_type else None
-        self.default_library = self._to_meson_shared(self._shared) \
-            if self._shared is not None else None
-        self.backend = self._to_meson_value(self._backend)
+        self._default_library = ("shared" if shared else "static") if shared is not None else None
+
+        compiler = (self._conanfile.settings.get_safe("compiler.base") or
+                    self._conanfile.settings.get_safe("compiler"))
+
+        cppstd = self._conanfile.settings.get_safe("compiler.cppstd")
+        self._cpp_std = self._to_meson_cppstd(compiler, cppstd) if cppstd else None
+
+        if compiler == "Visual Studio":
+            vscrt = self._conanfile.settings.get_safe("compiler.base.runtime") or \
+                    self._conanfile.settings.get_safe("compiler.runtime")
+            self._b_vscrt = {"MD": "md",
```

Comment:
that might use `.lower()`

Reply:
another note (might be slightly unrelated), in CCI users, have started to adopt MSVC compiler already, and now recipes need to use a fairly complicated logic to manage compiler runtime, as settings are very different between old and new compilers. do you have a plan to add some sort of helper, with can return uniform runtime values (e.g. in formats like `MD` or `MultiThreadedDebug`)?